### PR TITLE
Updates `gridplus-sdk`, Adds CRUD for ABI Contracts

### PR DESCRIPTION
**Pull request by douglance**
- Updates to v1.1.0 of `gridplus-sdk`
- Adds new `ContractTable` component for CRUD for ABI Contracts
- Adds new `useFeature` hook for managing feature flags
   - Updates `EthContracts` component to functional component to be able to use hooks 
- Swaps to Promise API for `gridplus-sdk`

### CODE REVIEW

File: src/components/AddressTagsPage.tsx [hunk 1]
```diff
@@ -52,8 +42,19 @@ const AddressTagsPage = ({
             setRetryFunction(fetchRecords);
           }
         });
+      if (res) {
```
Comment (alex-miller-0) [inline src/components/AddressTagsPage.tsx:45]: When would there not be `res`? Also you should check for `res.records` and other stuff, no?

File: src/components/ContractTable.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  };
+
+  const handleOnSelectAll = (_, _selectedContracts) => {
+    setSelectedContracts(_selectedContracts);
+  };
+
+  const removeSelected = () => {
+    setIsLoading(true);
+    const sigs = selectedContracts.map((contract) => contract.header.sig);
+    session.client.removeAbiRecords({ sigs }, (err, val) => {
```
Comment (alex-miller-0) [inline src/components/ContractTable.tsx:81]: Feels like SDK calls should not be in `.tsx` files... can you put this somewhere else?

File: src/components/ethContracts.tsx [hunk 3]
```diff
@@ -1,144 +1,87 @@
...
-    this.setState({ tab: key, success: false, error: null, loading: false });
+  function setDefaultState() {
+    setDefs([]);
+    setSuccess(false);
+    setLoading(false);
+    setCustomDefs([]);
+    setCustomDefsStr("");
   }
 
-  onSmartContractAddress(input) {
```
Comment (alex-miller-0) [inline src/components/ethContracts.tsx:54]: Not seeing this code replicated anywhere else... how are you responding to contract addresses now?

File: src/components/ethContracts.tsx [hunk 4]
```diff
@@ -1,144 +1,87 @@
...
-    this.setState({ loading: true, error: null });
-    // Longer timeout for loading these since requests may get dropped
-    this.props.session.client.timeout = 2 * constants.ASYNC_SDK_TIMEOUT;
-    const defs = this.state.customDefs
-      ? this.state.customDefs
-      : this.state.defs;
-    this.props.session.addAbiDefs(defs, (err) => {
+  function addDefs(skipErrors = false, defsToAdd = null) {
+    setError(null);
+    setLoading(true); // Longer timeout for loading these since requests may get dropped
```
Comment (alex-miller-0) [inline src/components/ethContracts.tsx:49]: Missing line break on this comment

File: src/components/ethContracts.tsx (hunk 3 above)
Comment (alex-miller-0) [reply src/components/ethContracts.tsx:54]: Maybe related, I put in a contract address and it found the functions. When I clicked to load them nothing happened and I see:

image

image

Comment (alex-miller-0) [reply src/components/ethContracts.tsx:54]: Same problem loading packs.

**Review by alex-miller-0 — CHANGES_REQUESTED** [review]
Contract loading is broke. Otherwise changes look good - a few minor comments.

File: src/components/ContractTable.tsx (hunk 2 above)
Comment (douglance) [reply src/components/ContractTable.tsx:81]: Removing all the SDK stuff from components would be a pretty big change. 

I have a branch where I pull out all of the session code and put it into a `useContext` that is made available to components with a custom hook. Do you want me to do it separately in that PR?

File: src/components/ethContracts.tsx (hunk 4 above)
Comment (douglance) [reply src/components/ethContracts.tsx]: [967d1d9](https://github.com/GridPlus/lattice-manager/pull/227/commits/967d1d92d5c90eefc8e2ab7978b6545f01559bde)

File: src/components/ethContracts.tsx (hunk 3 above)
Comment (douglance) [reply src/components/ethContracts.tsx:54]: [967d1d9](https://github.com/GridPlus/lattice-manager/pull/227/commits/967d1d92d5c90eefc8e2ab7978b6545f01559bde)

File: src/components/ContractTable.tsx (hunk 2 above)
Comment (alex-miller-0) [reply src/components/ContractTable.tsx:81]: Sorry thought I deleted this comment. Disregard.

File: src/components/ethContracts.tsx (hunk 3 above)
Comment (douglance) [reply src/components/ethContracts.tsx:54]: Oh and `onSmartContractAddress` was moved to [`SearchCard.tsx`](https://github.com/GridPlus/lattice-manager/blob/967d1d92d5c90eefc8e2ab7978b6545f01559bde/src/components/SearchCard.tsx#L30) but I forgot to remove it here

File: src/components/ContractTable.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+              dataSource={record.params}
+              pagination={false}
+              rowKey={(r) => r.name}
+            />
+          ),
+          rowExpandable: (record) => record.params.length > 0,
+        }}
+      >
+        <Table.Column
+          title="Name"
```
Comment (alex-miller-0) [inline src/components/ContractTable.tsx:164]: Please change to "Function Name"

File: src/components/ContractTable.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+          dataIndex={["header", "name"]}
+          defaultSortOrder="ascend"
+          sorter={(a: any, b: any) => a.header.name.localeCompare(b.val)}
+        />
+        <Table.Column
+          title="Sig"
```
Comment (alex-miller-0) [inline src/components/ContractTable.tsx:170]: I think "ID" might make more sense to the average user. Thoughts?

File: src/components/AddressTagsPage.tsx (hunk 1 above)
Comment (douglance) [reply src/components/AddressTagsPage.tsx:45]: `res` should always exist. You're right.

File: src/components/ContractTable.tsx (hunk 6 above)
Comment (douglance) [reply src/components/ContractTable.tsx]: I used the name "Identifier" 

[eea4e25](https://github.com/GridPlus/lattice-manager/pull/227/commits/eea4e25af17755c69d536b5c8ad1bcc0b971f258)

File: src/components/ContractTable.tsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+          onSelectAll: handleOnSelectAll,
+          selectedRowKeys: selectedContracts.map(
+            (contract) => contract?.header?.name
+          ),
+        }}
+        expandable={{
+          expandedRowRender: (record) => (
+            <Table
+              columns={[
+                { title: "Name", dataIndex: "name", key: "name" },
```
[the remark this answers is not in the document]
Comment (douglance) [reply src/components/ContractTable.tsx]: [eea4e25](https://github.com/GridPlus/lattice-manager/pull/227/commits/eea4e25af17755c69d536b5c8ad1bcc0b971f258)

File: src/components/ContractTable.tsx (hunk 5 above)
Comment (douglance) [reply src/components/ContractTable.tsx]: [eea4e25](https://github.com/GridPlus/lattice-manager/pull/227/commits/eea4e25af17755c69d536b5c8ad1bcc0b971f258)

**Review by alex-miller-0 — APPROVED** [review]

**Pull request merged**